src/main/java/com/bean00/Response.java

```diff
@@ -1,22 +1,99 @@
 package com.bean00;
 
+import java.util.HashMap;
+import java.util.Map;
+
 public class Response {
     private static final String PROTOCOL_VERSION = "HTTP/1.1";
+
+    private HashMap<String, Integer> headers = new HashMap<>();
     private int statusCode;
-    private String message;
+    private String body = "";
+    private int contentLength;
 
     public Response(int statusCode) {
         this.statusCode = statusCode;
-        this.message = Status.getMessage(statusCode);
+    }
+
+    public Response(int statusCode, byte[] rawBody) {
+        this(statusCode);
+        this.body = getBodyAsAString(rawBody);
+        this.contentLength = rawBody.length;
+        addHeaders();
     }
 
     @Override
     public String toString() {
-        return PROTOCOL_VERSION + " " + statusCode + " " + message + "\r\n";
+        String statusMessage = Status.getMessage(statusCode);
+
+        String response = buildResponse(statusMessage);
+
+        return response;
     }
 
     public int getStatusCode() {
         return statusCode;
     }
 
+    public String getBody() {
```

Comment:
Return a byte array instead, and convert it to a string in the tests where it's used

Reply:
Done